donation_base/wizard/tax_receipt_annual_create.py

```diff
@@ -0,0 +1,100 @@
+# © 2014-2016 Barroux Abbey (http://www.barroux.org)
+# © 2014-2016 Akretion France (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+from datetime import datetime
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class TaxReceiptAnnualCreate(models.TransientModel):
+    _name = 'tax.receipt.annual.create'
+    _description = 'Generate Annual Tax Receipts'
+
+    @api.model
+    def _default_end_date(self):
+        return datetime(datetime.today().year - 1, 12, 31)
```

Comment:
on functional review we found this issue:
datetime is no longer used. Instead it works with fields.Date. This need fixing

If you handle dates and datetimes, you might need to adapt your code now that Odoo returns always native Python objects, so no more fields.Date/Datetime.from_string is needed. For converting old style date/datetime strings to Python object, use fields.Date.to_date and fields.Datetime.to_datetime methods.